pkg/gitrepo/gitrepo.go

```diff
@@ -0,0 +1,111 @@
+package gitrepo
+
+import (
+	"errors"
+	"fmt"
+	"github.com/argoproj/argo-cd/v2/util/localconfig"
+	"io"
+	"k8s.io/apimachinery/pkg/util/json"
+	"os"
+	"path/filepath"
+)
+
+type RepoCtx struct {
+	Url   string `json:"url,omitempty"`
+	Alias string `json:"alias,omitempty"`
+}
+
+type RepoCtxCfg struct {
+	Default RepoCtx   `json:"default,omitempty"`
+	Repos   []RepoCtx `json:"repos,omitempty"`
+}
+
+const (
+	repoCtxFile    = "repoctx"
+	RepoDefaultCtx = "default"
+)
+
+var (
+	ErrNotFound     = errors.New("given alias doesn't exist")
+	ErrLoadCfgFile  = errors.New("failed to load config file")
+	ErrOverwriteCfg = errors.New("cannot overwrite config file")
+)
+
+func GetRepoCfgPath() (string, error) {
+	cfgDir, err := localconfig.DefaultConfigDir()
```

Comment:
I mentioned this in another comment. I think arlon specific files should reside in arlon specific config directory.

Reply:
> Any particular suggestions on where would the config dir be?
In the same root dir as the argocd config directory.

Reply:
done